File: .agent/skills/deployment-audit/scripts/audit.py
#!/usr/bin/env python3
import os
import sys
import argparse
import subprocess
from pathlib import Path

def run_command(command, cwd=None):
    try:
        result = subprocess.run(
            command,
            cwd=cwd,
            shell=True,
            check=True,
            stdout=subprocess.PIPE,
            stderr=subprocess.PIPE,
            text=True
        )
        return result.stdout.strip()
    except subprocess.CalledProcessError as e:
        return None

def check_git_repo():
    if not run_command("git rev-parse --is-inside-work-tree"):
        print("❌ Not a git repository.")
        sys.exit(1)

def find_garbage_files():
    # Find AppleDouble files (._*) and .DS_Store
    print("🔍 Scanning for garbage files...")
    
    # Get all files tracked by git
    tracked_files = run_command("git ls-files").split('\n')
    
    garbage = []
    for f in tracked_files:
        path = Path(f)
        if path.name.startswith("._") or path.name == ".DS_Store":
            garbage.append(f)
            
    return garbage

def check_gitignore():
    print("REVIEWING .gitignore...")
    if not os.path.exists(".gitignore"):
        print("❌ .gitignore not found!")
        return False
        
    with open(".gitignore", "r") as f:
        content = f.read()
        
    issues = []
    if "._*" not in content:
        issues.append("Missing '._*' (AppleDouble files)")
    if ".DS_Store" not in content:
        issues.append("Missing '.DS_Store'")
        
    if issues:
        print("⚠️  .gitignore gaps found:")
        for i in issues:
            print(f"   - {i}")
        return False
    else:
        print("✅ .gitignore looks good.")
        return True

def fix_garbage(garbage_files):
    if not garbage_files:
        print("✅ No garbage files to clean.")
        return

    print(f"🧹 Removing {len(garbage_files)} garbage files from git...")
    for f in garbage_files:
        run_command(f"git rm -f '{f}'")
        print(f"   Deleted: {f}")
    
    print("✅ Cleanup complete. Remember to commit these changes!")

def main():
    parser = argparse.ArgumentParser(description="Audit codebase for Vercel deployment issues.")
    parser.add_argument("--fix", action="store_true", help="Automatically remove garbage files from git.")
    args = parser.parse_args()

    check_git_repo()
    
    # 1. Check .gitignore
    check_gitignore()
    
    # 2. Find garbage
    garbage = find_garbage_files()
    
    if garbage:
        print(f"❌ Found {len(garbage)} garbage files tracked by git:")
        for g in garbage[:10]:
            print(f"   - {g}")
        if len(garbage) > 10:
            print(f"   ...and {len(garbage) - 10} more.")
            
        if args.fix:
            fix_garbage(garbage)
        else:
            print("\n❌ Audit FAILED. Run with --fix to clean up.")
            sys.exit(1)
    else:
        print("✅ No tracked garbage files found.")
        print("🚀 Codebase looks clean for deployment!")

if __name__ == "__main__":
    main()

File: .agent/skills/deployment-audit/scripts/audit.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

function runCommand(command) {
    try {
        return execSync(command, { encoding: 'utf8', stdio: ['pipe', 'pipe', 'ignore'] }).trim();
    } catch (e) {
        return null;
    }
}

function checkGitRepo() {
    if (!runCommand('git rev-parse --is-inside-work-tree')) {
        console.error('❌ Not a git repository.');
        process.exit(1);
    }
}

function findGarbageFiles() {
    console.log('🔍 Scanning for garbage files...');
    const trackedFiles = runCommand('git ls-files').split('\n').filter(Boolean);

    return trackedFiles.filter(file => {
        const basename = path.basename(file);
        return basename.startsWith('._') || basename === '.DS_Store';
    });
}

function checkGitignore() {
    console.log('REVIEWING .gitignore...');
    if (!fs.existsSync('.gitignore')) {
        console.error('❌ .gitignore not found!');
        return false;
    }

    const content = fs.readFileSync('.gitignore', 'utf8');
    const issues = [];

    if (!content.includes('._*')) issues.push("Missing '._*' (AppleDouble files)");
    if (!content.includes('.DS_Store')) issues.push("Missing '.DS_Store'");

    if (issues.length > 0) {
        console.warn('⚠️  .gitignore gaps found:');
        issues.forEach(issue => console.warn(`   - ${issue}`));
        return false;
    } else {
        console.log('✅ .gitignore looks good.');
        return true;
    }
}

function fixGarbage(garbageFiles) {
    if (garbageFiles.length === 0) {
        console.log('✅ No garbage files to clean.');
        return;
    }

    console.log(`🧹 Removing ${garbageFiles.length} garbage files from git...`);
    garbageFiles.forEach(file => {
        runCommand(`git rm -f "${file}"`);
        console.log(`   Deleted: ${file}`);
    });

    console.log('✅ Cleanup complete. Remember to commit these changes!');
}

function main() {
    const args = process.argv.slice(2);
    const shouldFix = args.includes('--fix');

    checkGitRepo();
    checkGitignore();

    const garbage = findGarbageFiles();

    if (garbage.length > 0) {
        console.error(`❌ Found ${garbage.length} garbage files tracked by git:`);
        garbage.slice(0, 10).forEach(g => console.error(`   - ${g}`));
        if (garbage.length > 10) console.error(`   ...and ${garbage.length - 10} more.`);

        if (shouldFix) {
            fixGarbage(garbage);
        } else {
            console.error('\n❌ Audit FAILED. Run with --fix to clean up.');
            process.exit(1);
        }
    } else {
        console.log('✅ No tracked garbage files found.');
        console.log('🚀 Codebase looks clean for deployment!');
    }
}

main();
